File: Activity2.1/app/src/main/java/com/mobiledevices/miguel/activity21/MainActivity.java
package com.mobiledevices.miguel.activity21;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.ButtonBarLayout;
import android.util.Log;
import android.view.View;
import android.widget.AdapterView;
import android.widget.Button;
import android.widget.ListView;
import android.widget.Toast;

import org.json.JSONArray;
import org.json.JSONException;

public class MainActivity extends AppCompatActivity implements JSONRequest.JSONRequestCallback {

    Button load;
    ListView friends;
    JSONArray friendsJSON;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        load = (Button) findViewById(R.id.button_load);
        friends = (ListView) findViewById(R.id.lv_friends);

        load.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                MainActivity.this.request(v);
            }
        });

        friends.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                Intent i = new Intent(getApplicationContext(), FriendsActivity.class);
                try{
                    i.putExtra("JSON", friendsJSON.getJSONObject(position).toString());
                }
                catch(JSONException e) {
                    e.printStackTrace();
                }
                startActivity(i);
            }
        });

    }

    public void request(View v) {
        JSONRequest r = new JSONRequest(this);
        r.execute("https://api.myjson.com/bins/j5huz");
    }

    public void done(JSONArray jsonArray) {

        try {
            for(int i = 0; i < jsonArray.length(); i++) {
                Toast.makeText(this, jsonArray.getJSONObject(i).getString("name"), Toast.LENGTH_SHORT).show();
            }
            friendsJSON = jsonArray;
            JSONAdapter jsonAdapter = new JSONAdapter(this, friendsJSON);
            friends.setAdapter(jsonAdapter);

        }
        catch(JSONException e) {
            e.printStackTrace();
        }
    }
}

File: Activity3.3/Activity_3.3/MainActivity.cs
using Android.App;
using Android.Widget;
using Android.OS;
using Org.Json;
using Android.Util;

namespace Activity_3._3
{
    [Activity(Label = "Activity3.3", MainLauncher = true, Icon = "@drawable/icon")]
    public class MainActivity : Activity, JSONRequest.JSONListener
    {
        Button b1;
        ListView lv1;

        public void Listen(JSONArray posts)
        {
            JSONAdapter jsonAdapter = new JSONAdapter(this, posts);
            lv1.Adapter = jsonAdapter;
        }

        protected override void OnCreate(Bundle bundle)
        {
            base.OnCreate(bundle);

            // Set our view from the "main" layout resource
            SetContentView(Resource.Layout.Main);

            b1 = FindViewById<Button>(Resource.Id.bt_main_getJson);
            lv1 = FindViewById<ListView>(Resource.Id.lv_mai_lv);

            b1.Click += delegate
            {
                JSONRequest jsonRequest = new JSONRequest(this);
                jsonRequest.Execute("http://jsonplaceholder.typicode.com/posts");
            };
        }
    }
}

